docs/go-configuration.md

```diff
@@ -4,27 +4,599 @@ title: Scope Go Agent configuration
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Adding agent metadata
+
+You can add custom agent metadata that will be appended to the default metadata that the agent collects (such as host information).
+
+Additionally, it is possible to set an environment variable as metadata value for a certain key, whose final value will be evaluated at runtime.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_METADATA="CustomKey1=MyValue, CustomKey2=MyValue2, CustomPath=\$PATH"
+```
+
+This is equivalent to: 
+```go
+    map[string]interface{}{
+		"CustomKey1": "MyValue",
+		"CustomKey2": "MyValue2",
+		"CustomPath": os.Getenv("PATH"),
+	}
+```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithMetadata(map[string]interface{}{
+        "CustomKey1": "MyValue",
+        "CustomKey2": "MyValue2",
+        "CustomPath": os.Getenv("PATH"),
+    })))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithMetadata(map[string]interface{}{
+		"CustomKey1": "MyValue",
+		"CustomKey2": "MyValue2",
+		"CustomPath": os.Getenv("PATH"),
+	})))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Select agent metadata as test configuration
+
+Yon can select metadata keys to be considered as relevant in the configuration of the test.
+
+By default, Scope Go agent adds the folowwing metadata keys to the test configuration:
+
+- **platform.name**: the result of calling `runtime.GOOS`
+- **platform.architecture**: depends of the current architecture, can be `X64`, `X86`, `Arm`, `Arm64`
+- **go.version**: the result of calling `runtime.Version()`
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_CONFIGURATION="CustomKey1, CustomKey2"
+```
+
+This is equivalent to: 
+```go
+    []string{"CustomKey1", "CustomKey2"}
+```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithConfigurationKeys([]string{"CustomKey1", "CustomKey2"})))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithConfigurationKeys([]string{"CustomKey1", "CustomKey2"})))
```

Comment:
This doesn't seem like runtime instrumentation

Reply:
ok